src/fnft_nsev.c

```diff
@@ -156,220 +177,366 @@ INT fnft_nsev(
+        case nse_discretization_2SPLIT2S:
+        case nse_discretization_2SPLIT3S:
+        case nse_discretization_2SPLIT4B:
+        case nse_discretization_2SPLIT3A:
+        case nse_discretization_2SPLIT3B:
+        case nse_discretization_2SPLIT4A:
+        case nse_discretization_2SPLIT6B:
+        case nse_discretization_2SPLIT6A:
+        case nse_discretization_2SPLIT8B:
+        case nse_discretization_2SPLIT5A:
+        case nse_discretization_2SPLIT5B:
+        case nse_discretization_2SPLIT8A:
+        case nse_discretization_2SPLIT7A:
+        case nse_discretization_2SPLIT7B:
+        case nse_discretization_4SPLIT4A:
+        case nse_discretization_4SPLIT4B:
+            break;
+        case nse_discretization_BO:
+        case nse_discretization_CF4_2:
+        case nse_discretization_CF4_3:
+        case nse_discretization_CF5_3:
+        case nse_discretization_CF6_4:
+        case nse_discretization_ES4:
+        case nse_discretization_TES4:
+            if (opts->bound_state_localization != nsev_bsloc_NEWTON && kappa == +1){
+                ret_code = E_INVALID_ARGUMENT(opts->bound_state_localization);
+                goto leave_fun;
+            }
+            break;
+        default: // Unknown discretization
+            return E_INVALID_ARGUMENT(opts->discretization);
     }
-    transfer_matrix = malloc(i*sizeof(COMPLEX));
-    if (transfer_matrix == NULL) {
-        ret_code = E_NOMEM;
-        goto release_mem;
+    
+    
+    upsampling_factor = nse_discretization_upsampling_factor(opts->discretization);
+    if (upsampling_factor == 0) {
+        ret_code = E_INVALID_ARGUMENT(opts->discretization);
+        goto leave_fun;
     }
+    D_effective = D * upsampling_factor; // upsampling_factor*D gives the effective number of samples
     
     // Determine step size
     const REAL eps_t = (T[1] - T[0])/(D - 1);
-
-    // Compute the transfer matrix
-    if (opts->normalization_flag)
-        W_ptr = &W;
-    ret_code = nse_fscatter(D, q, eps_t, kappa, transfer_matrix, &deg, W_ptr,
-        opts->discretization);
-    CHECK_RETCODE(ret_code, release_mem);
     
-    // Compute the continuous spectrum
-    if (contspec != NULL && M > 0) {
-        ret_code = tf2contspec(deg, W, transfer_matrix, T, D, XI, M,
-            contspec, opts);
-        CHECK_RETCODE(ret_code, release_mem);
+    normconsts_or_residues_reserve = normconsts_or_residues;
+    if (opts->richardson_extrapolation_flag == 1){
+        ds_type_opt = opts->discspec_type;
+        if (ds_type_opt == nsev_dstype_RESIDUES){
+            opts->discspec_type = nsev_dstype_BOTH;
+            normconsts_or_residues_reserve = malloc(*K_ptr*2 * sizeof(COMPLEX));
+            if (normconsts_or_residues_reserve == NULL) {
+                ret_code = E_NOMEM;
+                goto leave_fun;
+            }
+        }
     }
     
-    // Compute the discrete spectrum
-    if (kappa == +1 && bound_states != NULL) {
+    // Preprocess signal
```

Comment:
This comment is not very useful. It is clear what `nse_preprocess_signal` preprocesses the signal. Maybe: What is happening during preprocessing? Why is it needed?